src/main.rs

```diff
@@ -142,7 +142,7 @@ fn update_cache(cache: &Cache, archive_source: &str, quietly: bool) -> Result<()
 }
 
 /// Show file paths
-fn show_paths(config: &Config) {
+fn show_paths(custom_config_path: Option<&Path>, config: &Config) {
```

Comment:
This interface does not really stay in one level of abstraction: The `Config` is a really high level object whereas the `custom_config_path` is just directly the value of one of the paths.

I think we can fix this by tracking the config file location in the `Config` option directly. I want to slowly refactor the config system so that `Config` is kind of "the ultimate config used at runtime", which is the combined values of CLI, env, and config file. In that sense, writing a value that doesn't directly come from the contents of the config file is still reasonable.

You can add a new field `config_file_location: PathBuf` to `Config` and set it as part of `Config::load` - I will propose another change to `Config::load` in the next comment